pkg/api/v1/types.go

```diff
@@ -1987,34 +1989,90 @@ type SecretList struct {
 	Items []Secret `json:"items" description:"items is a list of secret objects; see http://releases.k8s.io/HEAD/docs/user-guide/secrets.md"`
 }
 
-// Type and constants for component health validation.
+type ComponentType string
+
+const (
+	ComponentAPIServer         ComponentType = "apiserver"
+	ComponentControllerManager ComponentType = "controller-manager"
+	ComponentScheduler         ComponentType = "scheduler"
+)
+
+type ComponentSpec struct {
+	Type           ComponentType `json:"type" description:"type of the component"`
+	LivenessProbe  *Probe        `json:"livenessProbe,omitempty" description:"periodic probe of component liveness"`
+	ReadinessProbe *Probe        `json:"readinessProbe,omitempty" description:"periodic probe of component readiness"`
+}
+
+type ComponentPhase string
+
+const (
+	ComponentPending    ComponentPhase = "Pending"
+	ComponentRunning    ComponentPhase = "Running"
+	ComponentTerminated ComponentPhase = "Terminated"
+)
+
 type ComponentConditionType string
 
-// These are the valid conditions for the component.
 const (
-	ComponentHealthy ComponentConditionType = "Healthy"
+	ComponentRunningHealthy    ComponentConditionType = "Healthy"
+	ComponentTerminatedCleanly ComponentConditionType = "Cleanly"
 )
 
 type ComponentCondition struct {
-	Type    ComponentConditionType `json:"type" description:"type of component condition, currently only Healthy"`
-	Status  ConditionStatus        `json:"status" description:"current status of this component condition, one of True, False, Unknown"`
-	Message string                 `json:"message,omitempty" description:"health check message received from the component"`
-	Error   string                 `json:"error,omitempty" description:"error code from health check attempt (if any)"`
+	Type    ComponentConditionType `json:"type" description:"type of condition: Pending, Running, or Terminated"`
+	Status  ConditionStatus        `json:"status" description:"status of the condition: True, False, or Unknown"`
+	Reason  string                 `json:"message,omitempty" description:"reason (brief) for the transition to this condition"`
```

Comment:
Update desc and usage to match https://github.com/kubernetes/kubernetes/blob/master/docs/devel/api-conventions.md#typical-status-properties